libraries/helpers.rb

```diff
@@ -8,17 +8,29 @@ class CloudPassage
     # windows_configuration_variables and linux_configuration_variables,
     # respectively
     class ConfigHelper
-      def initialize(agent_key, opt = {})
-        @agent_key = agent_key
-        initialize_params(opt)
-        @win_conf = WinConfig.new(opt)
-        @lin_conf = LinConfig.new(opt)
+      def initialize(opt = {})
+        # Chef::Log.warn("Initialization Data: #{opt}")
+        final_params = initialize_params(opt)
+        @win_conf = WinConfig.new(final_params)
+        @lin_conf = LinConfig.new(final_params)
       end
 
       def initialize_params(params = {})
-        params.each do |key, val|
-          instance_variable_set("@#{key}", val) unless params.empty?
+        final_config = blend_configs(params[:base_config],
+                                     params[:databag_config],
+                                     params[:encrypted_databag_config])
+        final_config.each do |key, val|
+          instance_variable_set("@#{key}", val)
```

Comment:
It's generally a good idea to stay away from the `instance_variable_(get|set)` methods if you can avoid it. Since this is just a key-value hash of stuff, and `#blend_configs` already returns a hash, you can probably just create a single instance variable hash and put all the data there.

```
@config = blend_configs(...)
```